sky_provider/operators/sky_task.py

```diff
@@ -0,0 +1,260 @@
+        super().__init__(
+            python_callable=run_sky_task_with_credentials,
+            # PythonVirtualenvOperator now uses uv by default if available,
+            # so we need to use --pre, as the Azure CLI has an issue with uv.
+            pip_install_options=['--pre'],
+            # We had an issue with the latest (prerelease) version of httpx,
+            # so pinning it to the latest stable version for now.
+            requirements=[skypilot_requirement, 'httpx==0.28.1'],
+            python_version='3.11',
+            # Needed for git to work from the virtualenv.
+            system_site_packages=True,
+            **kwargs,
+        )
+
+        self.base_path = base_path
+        self.yaml_path = yaml_path
+        self.git_branch = git_branch
+        self.credentials_override = credentials_override
+        self.envs_override = envs_override or {}
+        self.skypilot_version = skypilot_version
+
+    def execute(self, context):
+        api_server_endpoint = Variable.get('SKYPILOT_API_SERVER_ENDPOINT')
+        if not api_server_endpoint:
+            raise AirflowException(
+                'SKYPILOT_API_SERVER_ENDPOINT Variable is not set')
+
+        # Get credentials in the main Airflow environment,
+        # as it depends on Airflow providers not
+        # available later in the virtualenv.
+        credentials = self._get_credentials()
+
+        self.op_args = [
+            self.base_path,
+            self.yaml_path,
+            self.git_branch,
+            credentials,
+            self.envs_override,
+            api_server_endpoint,
+        ]
+        return super().execute(context)
+
+    def _get_credentials(self) -> ClusterCredentials:
+        """
+        Get credential artifacts from all configured cloud providers.
+
+        Returns:
+            ClusterCredentials with combined file_mounts and env_vars from all providers
+        """
+        if self.credentials_override is None:
+            return {'file_mounts': {}, 'env_vars': {}}
+
+        file_mounts = {}
+        env_vars = {}
+        for cloud_name, handler_class in CREDENTIAL_HANDLERS.items():
+            conn_id = self.credentials_override.get(
+                cloud_name)  # type: ignore[literal-required]
+            if conn_id:
+                handler = handler_class(str(conn_id))
+                cluster_credentials = handler.get_cluster_credentials()
+                file_mounts.update(cluster_credentials['file_mounts'])
+                env_vars.update(cluster_credentials['env_vars'])
+
+        return {'file_mounts': file_mounts, 'env_vars': env_vars}
+
+
+def run_sky_task_with_credentials(
+    base_path: str,
+    yaml_path: str,
+    git_branch: Optional[str],
+    credentials: ClusterCredentials,
+    envs_override: Dict[str, str],
+    api_server_endpoint: str,
+):
+    import os
+    import subprocess
+    import tempfile
+    import uuid
+
+    import yaml
+
```

Comment:
should we run `sky.check` before the launch, so that people now whether their cloud credentials (the credentials used by skypilot to launch clusters) are setup correctly.

Reply:
Good idea, let me add that.

Reply:
Actually never mind, turns out we don't export `check` somehow: 

```
>>> import sky
>>> sky.check
<module 'sky.check' from '/Users/kevin/git/skypilot/sky/check.py'>
```